tests/program-data/programs/program-data/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+use anchor_lang::prelude::*;
+
+declare_id!("Cum9tTyj5HwcEiAmhgaS7Bbj4UczCwsucrCkxRECzM4e");
+
+// TODO: Once anchor can deserialize program data, update this test.
```

Comment:
Is this comment still relevant?

Reply:
yes, we should have a test that checks that you can uses settings, program, program_data, and the upgrade authority together to set data. this use case was what got these issues created.

eventually we should have a test that can check the following
```
#[derive(Accounts)]
#[instruction(admin_data: u64)]
pub struct SetAdmin {
  upgrade_authority_address: Signer<'info>,
  #[account(has_one = programdata_address)
  program: Program<'info, <YOUR_PROGRAM>,
  #[account(has_opt = upgrade_authority_address)]
  programdata_address: ProgramData<'info>
}
```

(pending has_opt feature approval)

Reply:
program data in that comment was referring to the data of the Program (i.e. programdata_address), not the data of ProgramData.